segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -121,7 +129,18 @@
     private void logEnd() {
         Duration elapsed = timer.getElapsed();
         SLTS_READ_LATENCY.reportSuccessEvent(elapsed);
+        SLTS_NUM_CHUNKS_READ.reportSuccessValue(cntChunksRead.get());
         SLTS_READ_BYTES.add(length);
+        if (segmentMetadata.isStorageSystemSegment()) {
+            SLTS_SYSTEM_READ_LATENCY.reportSuccessEvent(elapsed);
+            SLTS_SYSTEM_NUM_CHUNKS_READ.reportSuccessValue(cntChunksRead.get());
+            SLTS_SYSTEM_READ_BYTES.add(length);
+        }
+        if (elapsed.toMillis() > 0) {
+            val bytesPerSecond = 1000 * length / elapsed.toMillis();
```

Comment:
If you have more than 2MB in `length`, this will overflow.

Reply:
fixed